patches/chrome-android-java-src-org-chromium-chrome-browser-toolbar-bottom-BottomToolbarVariationManager.java.patch

```diff
@@ -0,0 +1,67 @@
+diff --git a/chrome/android/java/src/org/chromium/chrome/browser/toolbar/bottom/BottomToolbarVariationManager.java b/chrome/android/java/src/org/chromium/chrome/browser/toolbar/bottom/BottomToolbarVariationManager.java
+index 5aca8d3e33981d352282b868dd2ffaaa32a45a74..278dd69cb5a0bcd7deb23e9fef68ff1ddc062b4c 100644
+--- a/chrome/android/java/src/org/chromium/chrome/browser/toolbar/bottom/BottomToolbarVariationManager.java
++++ b/chrome/android/java/src/org/chromium/chrome/browser/toolbar/bottom/BottomToolbarVariationManager.java
+@@ -41,9 +41,9 @@ public class BottomToolbarVariationManager {
+             return Variations.HOME_SEARCH_TAB_SWITCHER;
+         }
+         sVariation = FeatureUtilities.getBottomToolbarVariation();
+-        if (sVariation.equals(Variations.NONE)) {
++/*        if (sVariation.equals(Variations.NONE)) {
+             return Variations.HOME_SEARCH_TAB_SWITCHER;
+-        }
++        }*/
+         return sVariation;
+     }
+ 
+@@ -52,7 +52,7 @@ public class BottomToolbarVariationManager {
+      *         in the current variation.
+      */
+     public static boolean isShareButtonOnBottom() {
+-        return FeatureUtilities.isBottomToolbarEnabled()
++        return FeatureUtilities.isBottomToolbarEnabled() && !getVariation().equals(Variations.NONE)
+                 && !getVariation().equals(Variations.HOME_SEARCH_TAB_SWITCHER);
+     }
+ 
+@@ -62,7 +62,7 @@ public class BottomToolbarVariationManager {
+      */
+     public static boolean isNewTabButtonOnBottom() {
+         return FeatureUtilities.isBottomToolbarEnabled()
+-                && getVariation().equals(Variations.NEW_TAB_SEARCH_SHARE);
++                && (getVariation().equals(Variations.NEW_TAB_SEARCH_SHARE) || getVariation().equals(Variations.NONE));
```

Comment:
same as above if you drop the parens (&& has higher precedence, same as - https://github.com/brave/brave-core/pull/4491/files#diff-dfe64324850e34ad41ff404622c925a6R49)

Reply:
Dropping parens will change the logic here. Or you mean to add string `|| FeatureUtilities.isBottomToolbarEnabled() && getVariation().equals(Variations.NONE)` after `&& getVariation().equals(Variations.NEW_TAB_SEARCH_SHARE);`. But in this case I will have to remove semicolon, so that line will be patched anyway.
Is it ok if I add function `isNewTabButtonOnBottom` to `BraveBottomToolbarVariationManager` class (I will create it for `isBookmarkButtonOnBottom` either way). And patch will be `if (true) return BraveBottomToolbarVariationManager.isNewTabButtonOnBottom()` right before this return. Will it work for you?

Reply:
I think the patch would have to be `if (BraveBottomToolbarVariationManager. isNewTabButtonOnBottom) return true;` other wise anything calling `BottomToolbarVariationManager.isNewTabButtonOnBottom` would get the wrong value

Reply:
I think just leave this one for now